src/robot/running/namespace.py

```diff
@@ -338,6 +339,42 @@ def _get_runner_from_libraries(self, name):
             return found[0]
         self._raise_multiple_keywords_found(name, found)
 
+    def _handle_private_runners(self, runners):
+        sum_public = sum("robot:private" not in runner.tags for runner in runners)
+        sum_private = sum("robot:private" in runner.tags for runner in runners)
+
+        if sum_public == len(runners) or sum_private == len(runners) or sum_public > 1:
+            return runners
+
+        for runner in runners:
+            if 'robot:private' not in runner.tags:
+                public_runner = runner
+                private_runners = runners.copy()
+                private_runners.remove(public_runner)
+                break
```

Comment:
Probably we could initially filter runners so that we have list of public and private runners instead of `sum_xxx` containing just their counts. Then this loop wouldn't needed at all.